src/function/built_in_functions.cpp

```diff
@@ -68,6 +68,9 @@ Function* BuiltInFunctions::matchScalarFunction(
     uint32_t minCost = UINT32_MAX;
     for (auto& function : functionSet) {
         auto func = reinterpret_cast<Function*>(function.get());
+        if (name == CAST_FUNC_NAME) {
```

Comment:
Can we remove this. And inside `bindScalarFunctionExpression` we check functionName == "Cast" and directly get the function without going through the the `matchScalarFunction`